utils/duplicate/archiver.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from copy import deepcopy
+from typing import Tuple
+
+from requests.exceptions import HTTPError
+from assertpy import assert_warn
+from biosamples_v4.api import Client as BioSamplesClient
+from ingest.api.ingestapi import IngestApi
+from submission_broker.services.biosamples import BioSamples
+
+from converter.biosamples import BioSamplesConverter
+from .loader import DuplicateLoader
+from .submission import DuplicateSubmission, HcaSubmission
+
+
+class DuplicateArchiver:
+    def __init__(self, ingest: IngestApi, biosamples: BioSamplesClient, test_biosamples: BioSamples, converter: BioSamplesConverter):
+        self.ingest = ingest
+        self.loader = DuplicateLoader(ingest)
+        self.read_biosamples = biosamples
+        self.converter = converter
+        self.write_biosamples = test_biosamples
+        logger = logging.getLogger('biosample_comparisons')
+        logger.setLevel(logging.INFO)
+        fh = logging.FileHandler("biosample_comparisons.log")
+        fh.setLevel(logging.INFO)
+        logger.addHandler(fh)
+
+    def compare_duplicate_biosample(self, biosamples_accession: str, ignored_keys):
+        submission, biomaterial_uuid, project_uuid = self.__load_project_and_biomaterial(biosamples_accession)
+        self.__send_biosample(submission, biomaterial_uuid, project_uuid)
+        self.__log_comparison(submission, biomaterial_uuid, ignored_keys)
+
+    def compare_duplicate_project(self, project_uuid: str, ignored_keys):
+        submission = self.loader.duplicate_project(project_uuid)
+        for biomaterial_uuid, biosamples_accession in submission.old_accessions['BioSamples'].items():
+            biosample = self.read_biosamples.fetch_sample(biosamples_accession)
+            submission.biosamples.setdefault(biomaterial_uuid, {})['old'] = biosample
+            self.__send_biosample(submission, biomaterial_uuid, project_uuid)
+            self.__log_comparison(submission, biomaterial_uuid, ignored_keys)
+
+    def __load_project_and_biomaterial(self, biosamples_accession: str) -> Tuple[DuplicateSubmission, str, str]:
```

Comment:
There are lots of low level details in this function. Consider extracting some higher level functions that encapsulate the steps in the process.

Reply:
Indeed, this method and `__send_biosample` are both rewrites of the functionality we are testing and should be removed long-term.
This method only loads the specified project and biosamples without loading any of the other entities.
This is crucial while testing, or trying different mappings because its takes 10 minutes or more to load the entire project from ingest whereas running this method gets you feedback with in a few seconds.

Reply:
I've updated the comments around these methods to make these tradeoffs more explicit.

Reply:
I don't know what you mean, but I've added comments around these methods to explain the intention / tradeoffs with investing more into tidying this up.